pki/pki.go

```diff
@@ -480,176 +635,223 @@ type caDefaults struct {
 		AuthorityConfig: &authconfig.AuthConfig{
 			Options:              authorityOptions,
 			DisableIssuedAtCheck: false,
-			Provisioners:         provisioner.List{prov},
-		},
-		TLS: &authconfig.TLSOptions{
-			MinVersion:    authconfig.DefaultTLSMinVersion,
-			MaxVersion:    authconfig.DefaultTLSMaxVersion,
-			Renegotiation: authconfig.DefaultTLSRenegotiation,
-			CipherSuites:  authconfig.DefaultTLSCipherSuites,
+			EnableAdmin:          false,
 		},
+		TLS:       &authconfig.DefaultTLSOptions,
 		Templates: p.getTemplates(),
 	}
-	if p.enableSSH {
-		enableSSHCA := true
-		config.SSH = &authconfig.SSHConfig{
-			HostKey: p.sshHostKey,
-			UserKey: p.sshUserKey,
+
+	// Add linked as a deployment type to detect it on start and provide a
+	// message if the token is not given.
+	if p.options.deploymentType == LinkedDeployment {
+		config.AuthorityConfig.DeploymentType = LinkedDeployment.String()
+	}
+
+	// On standalone deployments add the provisioners to either the ca.json or
+	// the database.
+	var provisioners []provisioner.Interface
+	if p.options.deploymentType == StandaloneDeployment {
+		key, err := p.ottPrivateKey.CompactSerialize()
+		if err != nil {
+			return nil, errors.Wrap(err, "error serializing private key")
+		}
+
+		prov := &provisioner.JWK{
+			Name:         p.options.provisioner,
+			Type:         "JWK",
+			Key:          p.ottPublicKey,
+			EncryptedKey: key,
 		}
-		// Enable SSH authorization for default JWK provisioner
-		prov.Claims = &provisioner.Claims{
-			EnableSSHCA: &enableSSHCA,
+		provisioners = append(provisioners, prov)
+
+		// Add default ACME provisioner if enabled
+		if p.options.enableACME {
+			provisioners = append(provisioners, &provisioner.ACME{
+				Type: "ACME",
+				Name: "acme",
+			})
 		}
-		// Add default SSHPOP provisioner
-		sshpop := &provisioner.SSHPOP{
-			Type: "SSHPOP",
-			Name: "sshpop",
-			Claims: &provisioner.Claims{
+
+		if p.options.enableSSH {
+			enableSSHCA := true
+			config.SSH = &authconfig.SSHConfig{
+				HostKey: p.Ssh.HostKey,
+				UserKey: p.Ssh.UserKey,
+			}
+			// Enable SSH authorization for default JWK provisioner
+			prov.Claims = &provisioner.Claims{
 				EnableSSHCA: &enableSSHCA,
-			},
+			}
+
+			// Add default SSHPOP provisioner
+			provisioners = append(provisioners, &provisioner.SSHPOP{
+				Type: "SSHPOP",
+				Name: "sshpop",
+				Claims: &provisioner.Claims{
+					EnableSSHCA: &enableSSHCA,
+				},
+			})
 		}
-		config.AuthorityConfig.Provisioners = append(config.AuthorityConfig.Provisioners, sshpop)
 	}
 
 	// Apply configuration modifiers
 	for _, o := range opt {
-		if err = o(config); err != nil {
+		if err := o(config); err != nil {
 			return nil, err
 		}
 	}
 
+	// Set authority.enableAdmin to true
+	if p.options.enableAdmin {
+		config.AuthorityConfig.EnableAdmin = true
+	}
+
+	if p.options.deploymentType == StandaloneDeployment {
+		if !config.AuthorityConfig.EnableAdmin {
+			config.AuthorityConfig.Provisioners = provisioners
+		} else {
```

Comment:
Let's add a comment here - this section won't be used until we parameterize databases for `step ca init`.

Reply:
Fixed with 516b74f